diff --git a/registration/admin.py b/registration/admin.py
index 6541f6a..ef6da3b 100644
--- a/registration/admin.py
+++ b/registration/admin.py
@@ -1,6 +1,9 @@
 from django.contrib import admin
-from django.contrib.sites.models import RequestSite
 from django.contrib.sites.models import Site
+try:
+    from django.contrib.sites.requests import RequestSite
+except ImportError:
+    from django.contrib.sites.models import RequestSite
 from django.utils.translation import ugettext_lazy as _
 
 from registration.models import RegistrationProfile
diff --git a/registration/auth_urls.py b/registration/auth_urls.py
index 4e26d5f..1890fd7 100644
--- a/registration/auth_urls.py
+++ b/registration/auth_urls.py
@@ -24,14 +24,13 @@ consult a specific backend's documentation for details.
 """
 
 from django.conf.urls import include
-from django.conf.urls import patterns
 from django.conf.urls import url
 from django.core.urlresolvers import reverse_lazy
 
 from django.contrib.auth import views as auth_views
 
 
-urlpatterns = patterns('',
+urlpatterns = [
                        url(r'^login/$',
                            auth_views.login,
                            {'template_name': 'registration/login.html'},
@@ -61,4 +60,4 @@ urlpatterns = patterns('',
                        url(r'^password/reset/done/$',
                            auth_views.password_reset_done,
                            name='auth_password_reset_done'),
-)
+]
diff --git a/registration/backends/default/urls.py b/registration/backends/default/urls.py
index 3321c9d..c814fb6 100644
--- a/registration/backends/default/urls.py
+++ b/registration/backends/default/urls.py
@@ -17,7 +17,6 @@ your own URL patterns for these views instead.
 """
 
 
-from django.conf.urls import patterns
 from django.conf.urls import include
 from django.conf.urls import url
 from django.views.generic.base import TemplateView
@@ -26,7 +25,7 @@ from registration.backends.default.views import ActivationView
 from registration.backends.default.views import RegistrationView
 
 
-urlpatterns = patterns('',
+urlpatterns = [
                        url(r'^activate/complete/$',
                            TemplateView.as_view(template_name='registration/activation_complete.html'),
                            name='registration_activation_complete'),
@@ -46,5 +45,5 @@ urlpatterns = patterns('',
                        url(r'^register/closed/$',
                            TemplateView.as_view(template_name='registration/registration_closed.html'),
                            name='registration_disallowed'),
-                       (r'', include('registration.auth_urls')),
-                       )
+                       url(r'', include('registration.auth_urls')),
+                       ]
diff --git a/registration/backends/default/views.py b/registration/backends/default/views.py
index 4524152..dae7ee8 100644
--- a/registration/backends/default/views.py
+++ b/registration/backends/default/views.py
@@ -1,6 +1,9 @@
 from django.conf import settings
-from django.contrib.sites.models import RequestSite
 from django.contrib.sites.models import Site
+try:
+    from django.contrib.sites.requests import RequestSite
+except ImportError:
+    from django.contrib.sites.models import RequestSite
 
 from registration import signals
 from registration.models import RegistrationProfile
diff --git a/registration/backends/simple/urls.py b/registration/backends/simple/urls.py
index 0be6ee2..c01077b 100644
--- a/registration/backends/simple/urls.py
+++ b/registration/backends/simple/urls.py
@@ -18,19 +18,18 @@ your own URL patterns for these views instead.
 
 
 from django.conf.urls import include
-from django.conf.urls import patterns
 from django.conf.urls import url
 from django.views.generic.base import TemplateView
 
 from registration.backends.simple.views import RegistrationView
 
 
-urlpatterns = patterns('',
+urlpatterns = [
                        url(r'^register/$',
                            RegistrationView.as_view(),
                            name='registration_register'),
                        url(r'^register/closed/$',
                            TemplateView.as_view(template_name='registration/registration_closed.html'),
                            name='registration_disallowed'),
-                       (r'', include('registration.auth_urls')),
-                       )
+                       url(r'', include('registration.auth_urls')),
+                       ]
diff --git a/registration/models.py b/registration/models.py
index cf1cdbd..75ceed1 100644
--- a/registration/models.py
+++ b/registration/models.py
@@ -88,7 +88,7 @@ class RegistrationManager(models.Manager):
             registration_profile.send_activation_email(site)
 
         return new_user
-    create_inactive_user = transaction.commit_on_success(create_inactive_user)
+    create_inactive_user = transaction.atomic(create_inactive_user)
 
     def create_profile(self, user):
         """
